src/components/Button.tsx

```diff
@@ -25,6 +24,7 @@ interface ButtonProps {
   customTextStyle?: TextStyle
   testID?: string
   hasRightArrow?: boolean
+  hasPlusIcon?: boolean
```

Comment:
I see why you are doing this, but it makes me a bit sad.

Reply:
What alternative would you suggest?

Reply:
dont use a common button component. But being a bit glib there, there is not a good general button solution that works in all cases in RN as far as i know and what you have here works!

this would belong in a future commit, but you could allow the button to accept children which could include an icon. Or perhaps you have a `rightIcon` prop that can be either an arrow or a plus. or continue with the pattern you are establishing here. I think all the patterns have their pros and cons, tbh. My main sadness is from the accumulated complexity that all projects have.

Reply:
Yeah I get that! Let's pair on cleaning this up soon.